dbschema/seeds/011.product.edgeql

```diff
@@ -0,0 +1,1080 @@
+      "progressTarget": 100,
+      "progressStepMeasurement":"Percent",
+      "totalVerses": 25,
+      "totalVerseEquivalents": 25
+    },
+    {
+      "project": "Lothlorien",
+      "unspecifiedScripture":
+        {
+          "book": "Exodus",
+          "totalVerses": 22
+        },
+      "steps": [
+          "InternalizationAndDrafting",
+          "PeerRevision",
+          "CommunityTesting",
+          "BackTranslation",
+          "ConsultantCheck",
+          "Completed"
+        ],
+      "methodology": "Render",
+      "describeCompletion": "Moses call-Ex 3:1-22",
+      "pnpIndex": 64,
+      "progressTarget": 100,
+      "progressStepMeasurement":"Percent",
+      "totalVerses": 22,
+      "totalVerseEquivalents": 22
+    }
+    ]'),
+     products := (
+        for directScriptureProduct in json_array_unpack(productsJson)
+        union (
+          (select DirectScriptureProduct filter .project.name = <str>directScriptureProduct['project']
+           and (.scripture.label ?= <str>json_get(directScriptureProduct, 'scripture', 'label')
+                or .unspecifiedScripture.book ?= <str>json_get(directScriptureProduct, 'unspecifiedScripture', 'book'))) ??
+          (with
+            scripture := if (select exists json_get(directScriptureProduct,'scripture')) then (
+              insert Scripture::Collection {
+                  label := <str>directScriptureProduct['scripture']['label'],
+                  verses := (
+                      for verseRange in json_array_unpack(directScriptureProduct['scripture']['verses'])
+                      union (
+                          insert Scripture::VerseRange {
+                              label := <str>verseRange['label'],
+                              `start` := (
+                                  insert Scripture::Verse {
+                                      book := <str>verseRange['start']['book'],
+                                      chapter := <int16>verseRange['start']['chapter'],
+                                      verse := <int16>verseRange['start']['verse'],
+                                      verseId := <int16>verseRange['start']['verseId']
+                                  }
+                              ),
+                              `end` := (
+                                  insert Scripture::Verse {
+                                      book := <str>verseRange['end']['book'],
+                                      chapter := <int16>verseRange['end']['chapter'],
+                                      verse := <int16>verseRange['end']['verse'],
+                                      verseId := <int16>verseRange['end']['verseId']
+                                  }
+                              )
+                          }
+                      )
+                  )
+              })
+              else {},
+            engagement := assert_single((select Engagement filter .project.name = <str>directScriptureProduct['project']))
```

Comment:
I'd move this to a with statement above the select/insert. This is a datapoint we are expecting to pre-exist as a dependency to this script.
Modeled here: https://github.com/SeedCompany/cord-api-v3/blob/develop/dbschema/seeds/009.language-engagements.edgeql#L37-L42

With that scoped you can change your `select` check to
```edgeql
select DirectScriptureProduct filter .project = engagement.project
```

This will make it easier to adapt this to support multiple engagements within a project. Something we should do soon.

Reply:
Actually the select should be
```edgeql
select DirectScriptureProduct filter .engagement = engagement
```